modules/peers.js

```diff
@@ -710,6 +710,7 @@ Peers.prototype.list = function(options, cb) {
 					// Unmatched broadhash
 					return randomList(peers, waterCb);
 				}
+				peers = __private.shuffle(peers);
```

Comment:
Why we shuffle peers again here?

Reply:
@4miners Before, it would concatenate the shuffled list of matched broadhash peers with the shuffled list of unmatched peers - This means that the matched broadhash peers would always be at the top of the list and would always be in the top 20 when choosing which peers to broadcast to (because we use array slice). So it's like both halves of the final list were shuffled but the whole list wasn't actually shuffled.

It seems like the previous behaviour was unintentional but I can roll back if necessary.

In any case, we don't need both both. I didn't want to remove the first shuffle in the getByFilter function because it appears to be used in other parts of the code.